daemon/k8s_watcher.go

```diff
@@ -685,6 +685,11 @@ func (d *Daemon) updateK8sNetworkPolicyV1(oldk8sNP, newk8sNP *networkingv1.Netwo
 func (d *Daemon) deleteK8sNetworkPolicyV1(k8sNP *networkingv1.NetworkPolicy) {
 	labels := k8s.GetPolicyLabelsv1(k8sNP)
 
+	if labels == nil {
+		log.Debugf("provided v1 NetworkPolicy is nil, so cannot delete it")
```

Comment:
When does this happen? The problem with Debug and silent return is that it will never be visible in any non-developer environment. I'm fine to explicitly panic here instead of having the dereference cause it.